src/commissaire_service/storage/__init__.py

```diff
@@ -0,0 +1,267 @@
+# Copyright (C) 2016  Red Hat, Inc
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import fnmatch
+import importlib
+import json
+
+from commissaire import constants as C
+
+from commissaire_service.service import CommissaireService, ServiceManager
+from commissaire_service.storage.storehandlermanager import (
+    StoreHandlerManager, ConfigurationError)
+
+
+# FIXME This should be moved to a place for utilities.
+def read_config_file(path=None):
+    """
+    Attempts to parse a configuration file, formatted as a JSON object.
+
+    If a config file path is explicitly given, then failure to open the
+    file will raise an IOError.  Otherwise a default path is tried, but
+    no IOError is raised on failure.  If the file can be opened but not
+    parsed, an exception is always raised.
+
+    :param path: Full path to the config file, or None
+    :type path: str or None
+    :returns: configuration content as a dictionary
+    :rtype: dict
+    :raises: IOError, TypeError, ValueError
+    """
+    json_object = {}
+    using_default = False
+
+    if path is None:
+        path = '/etc/commissaire/commissaire.conf'
+        using_default = True
+
+    try:
+        with open(path, 'r') as fp:
+            json_object = json.load(fp)
+        if using_default:
+            print('Using configuration in {0}'.format(path))
+    except IOError:
+        if not using_default:
+            raise
+
+    if type(json_object) is not dict:
+        raise TypeError('{0}: '.format(path) +
```

Comment:
super anal: I just thought the line break there looked nicer, but I can change it.